### ISSUE TITLE
CPU spikes and loss of rules

### ISSUE THREAD

**Issue by Tyxz** [henrypp/simplewall#387]
I noticed recently two bug:
First the most annoying one: Could not reproduce yet, but sometimes simplewall has these huge CPU usage spikes. It literally crashes my whole system by demanding all my CPU power. Very annoying, because only a cold reboot can kill the process.
The closest I came to reproduce is, when I exit simplewall and then reopen it, it happens after a while.

The second one is a loss of all my rules after the last update (I think).

Second one is not a huge issue, but first one is sadly a reason for me to uninstall your otherwise great software for now.

**Comment by perdolka** [comment]
> loss of all my rules

I wrote about this rare bug too:
https://github.com/henrypp/simplewall/issues/378#issuecomment-481291932
https://github.com/henrypp/simplewall/issues/168

**Comment by Dark-Noir** [comment]
Yeah had the same issue several times.

**Comment by xnoreq** [comment]
I have lost all my rules twice. I think it happened after resuming from sleep/standby. See #391.

I have also seen the CPU load bug, but only once. I think it happened when simplewall opened the "allow/block/block for now" popup throwing me back to the desktop while I was in a fullscreen application.. That's also very annoying.

**Comment by haluzpav** [comment]
I also have the CPU usage bug. Happens about once a day. The Simplewall is still responsive, but the only fix is to kill the program. Disable filtering does not help.

**Comment by bfgh012** [comment]
Having the high CPU usage bug too, it just happens randomly, and you start to feel it when the system start to get slower.
The only way to stop it,  kill the process before the whole system freezes.

**Comment by medack** [comment]
Got the CPU bug also. I think I got a way to reproduce it (Windows 10 x64, simplewall 2.4.6):

1 - Open a new electron app.
2 - Once you get the popup to block or allow connection, block it.

These steps don't make much sense, but I managed to get simplewall to crash my system twice with two different electron apps this way. Don't know if this is linked to any of the other reports, but at least some of the symptoms are indeed similar.
